### DISCUSSION TITLE
Is it possible to transition inside a activity? (I heard it's an anti-pattern, I'm providing an example)

### DISCUSSION THREAD

**Discussion by ghost** [statelyai/xstate#1218] [General] [answered]
[Code here](https://codesandbox.io/s/xstate-question-7ydxz) _(please open the console)_

It's like a drag 'n drop machine, but in the 'dragging' state we start an activity that listens for pointerup events that happen outside of a target element. When we leave the dragging state, the activity stops and the listener is removed.

Is it possible to trigger the transition from dragging to idle inside the activity listener?

**Comment by Andarist** [comment, marked as answer]
You can use callback service for this - just send an event from it to the invoking machine so it can transition how you tell it to

https://codesandbox.io/s/xstate-question-2suxj?file=/src/index.tsx

**Reply by ghost** [reply]
Oh, didn't think of making use of services for this... Thank you!
